addons/mail/static/src/components/chat_window/chat_window.js

```diff
@@ -53,6 +53,8 @@ class ChatWindow extends Component {
     mounted() {
         this.env.messagingBus.on('will_hide_home_menu', this, this._onWillHideHomeMenu.bind(this));
         this.env.messagingBus.on('will_show_home_menu', this, this._onWillShowHomeMenu.bind(this));
+        this.env.messagingBus.on('hide_home_menu', this, this._onHideHomeMenu.bind(this));
+        this.env.messagingBus.on('show_home_menu', this, this._onShowHomeMenu.bind(this));
```

Comment:
Didn't find any models doing event listening so in which methods do you put .on and .off ?

Reply:
In `mail.chat_window`, .on on creation and .off on deletion.

Reply:
Main entry/exit points of models are static create and delete

Reply:
I used `init` and `delete`

Reply:
I should work fine but I'm not sure what's the policy, I think we prefer create instead of init.

init was introduced when there was pre-processing that was necessary before assigning value to fields (for example because assigning fields would need these added information, such as relying on them for computing onchange), but in this case, it is just binding events, the rest of the code will work fine if it is not done immediately. It it actually safer if the events are received only after the record is fully created.

Reply:
How do you access the env in static method ? through created record ?

Reply:
Did something that seems to work